### DISCUSSION TITLE
Unable to connect to the server: getting credentials: exec: exit status 1

### DISCUSSION THREAD

**Discussion by Strongeyed** [swellstores/community#274] [Help]
Hi team,

Got this error message when trying to push my store for development or production:

**Unable to connect to the server: getting credentials: exec: exit status 1**
remote: Error: failed to fetch credentials for cluster "xx": Unable to authenticate you

remote:        ERROR  Swell: unable to loading settings (Error: You must authenticate with your public key. Also, make sure withCredentials is enabled for all requests.)
remote:        FATAL  You must authenticate with your public key. Also, make sure withCredentials is enabled for all requests.

I have not made any changes, so was wondering if anyone is having similar issues?

**Comment by sags95** [comment]
Hi @Robin-DG , can you describe where you're seeing this error?

**Reply by Strongeyed** [reply]
I reckon it's actually on the digital ocean server side as this is part of the error message: .k8s.ondigitalocean.com/api?timeout=32s

```
remote: Error: failed to fetch credentials for cluster "xx": Unable to authenticate you
remote: error: unable to recognize "STDIN": Get https://xx.k8s.ondigitalocean.com/api?timeout=32s: getting credentials: exec: exit status 1



```

**Reply by mewa** [reply]
We've also been unable to git push, looks like the cluster has been decommissioned.

```
$ git push
Enumerating objects: 1, done.
Counting objects: 100% (1/1), done.
Writing objects: 100% (1/1), 182 bytes | 182.00 KiB/s, done.
Total 1 (delta 0), reused 0 (delta 0), pack-reused 0
remote: -----> Deploying branch: production
remote: 
remote: Error: failed to fetch credentials for cluster "6e0c2d6d-c5bc-4dac-8983-bc7273bcdceb": Unable to authenticate you
remote: Error: failed to fetch credentials for cluster "6e0c2d6d-c5bc-4dac-8983-bc7273bcdceb": Unable to authenticate you
remote: Error: failed to fetch credentials for cluster "6e0c2d6d-c5bc-4dac-8983-bc7273bcdceb": Unable to authenticate you
remote: Error: failed to fetch credentials for cluster "6e0c2d6d-c5bc-4dac-8983-bc7273bcdceb": Unable to authenticate you
remote: Error: failed to fetch credentials for cluster "6e0c2d6d-c5bc-4dac-8983-bc7273bcdceb": Unable to authenticate you
remote: Unable to connect to the server: getting credentials: exec: exit status 1
remote: -----> Preparing environment...
remote:        
remote:        ✅
remote:        
remote: -----> Installing dependencies...
remote:        
remote:        yarn install v1.22.19
remote:        [1/4] Resolving packages...
remote:        [2/4] Fetching packages...
remote:        [3/4] Linking dependencies...
remote:        warning "@nuxtjs/eslint-config > eslint-config-standard@16.0.3" has incorrect peer dependency "eslint@^7.12.1".
remote:        warning "@nuxtjs/eslint-config > eslint-config-standard@16.0.3" has incorrect peer dependency "eslint-plugin-promise@^4.2.1 || ^5.0.0".
remote:        [4/4] Building fresh packages...
remote:        $ husky install
remote:        fatal: not a git repository (or any of the parent directories): .git
remote:        Done in 50.65s.
remote:        
remote:        
remote:        
remote:        ✅
remote:        
remote: -----> Building app...
remote:        
remote:        yarn run v1.22.19
remote:        $ nuxt generate
remote:        ℹ Using Tailwind CSS from ~/assets/css/tailwind.css
remote:        ℹ Merging Tailwind config from ~/tailwind.config.js
remote:        
remote:        ERROR  Swell: unable to loading settings (Error: You must authenticate with your public key. Also, make sure withCredentials is enabled for all requests.)
remote:        
remote:        
remote:        FATAL  You must authenticate with your public key. Also, make sure withCredentials is enabled for all requests.
```

**Reply by logeshswell** [reply]
@mewa Please help me with your store-id. I can fix the missing env variable (`SWELL_PUBLIC_KEY`) in your storefront.

**Comment by Strongeyed** [comment]
Hi Axel,Sure, it’s when I do a $git push (once trying to deploy) on the production side. I’ve checked the GitHub key and it responds when using $ ssh -T ***@***.*** issue might lie with the swell store public key? Just letting you know it used to work without issues, I have not changed anything on my end.Kind Regards,Robin de GeusOn 17/06/2023, at 08:21, Alex Sagel ***@***.***> wrote:﻿
Hi @Robin-DG , can you describe where you're seeing this error?

**Comment by logeshswell** [comment]
> remote: FATAL You must authenticate with your public key. Also, make sure withCredentials is enabled for all requests.

@Strongeyed One of the reasons for getting this error is, either `SWELL_PUBLIC_KEY` env is missing or containing an invalid value.

**Reply by Strongeyed** [reply]
Hi @logeshswell this $git push is done to the Swell remote repository though, previously this has been no problem.

**Reply by logeshswell** [reply]
Yes, we are noticing this type of issue randomly in very few stores, after our storefront upgrade.
Can you help me with your store-id, so that I can fix it on our end?

**Reply by Strongeyed** [reply]
No problem @logeshswell store id is vGfMtsXq

**Reply by logeshswell** [reply]
@Strongeyed Can you try pushing your code change now?
If it still didn't work, please merge the latest origin theme code changes to your branch and then push it back. The publishing issue would be resolved.
